config/routes.rb

```diff
@@ -0,0 +1,9 @@
+Rails.application.routes.draw do
+  root 'tasks#index'
+  resources :tasks
+
+  patch '/tasks/:id/mark_complete', to: 'tasks#mark_complete', as: 'mark_complete'
+  patch '/tasks/:id/undo_mark_complete', to: 'tasks#undo_mark_complete', as: 'undo_mark_complete'
```

Comment:
You probably don't need two paths for this. One method could do the work of both.